var/spack/repos/builtin/packages/rocm-openmp-extras/package.py

```diff
@@ -471,13 +471,18 @@ def install(self, spec, prefix):
             os.unlink(os.path.join(bin_dir, "flang1"))
         if os.path.islink((os.path.join(bin_dir, "flang2"))):
             os.unlink(os.path.join(bin_dir, "flang2"))
+        if os.path.islink((os.path.join(bin_dir, "flang-legacy"))):
+            os.unlink(os.path.join(bin_dir, "flang-legacy"))
         if os.path.islink((os.path.join(lib_dir, "libdevice"))):
             os.unlink(os.path.join(lib_dir, "libdevice"))
         if os.path.islink((os.path.join(llvm_prefix, "lib-debug"))):
             os.unlink(os.path.join(llvm_prefix, "lib-debug"))
 
         os.symlink(os.path.join(omp_bin_dir, "flang1"), os.path.join(bin_dir, "flang1"))
         os.symlink(os.path.join(omp_bin_dir, "flang2"), os.path.join(bin_dir, "flang2"))
+        os.symlink(
```

Comment:
on 5.7.1, i see a link getting created inside install prefix for llvm-amdgpu/bin and when i execute like below-  /home/spack/opt/spack/linux-rhel9-skylake_avx512/gcc-11.3.1/rocm-openmp-extras-5.7.1-owmri3xcla2f65bth3gfxw5g52m4j7ke/bin/flang-legacy
bash: /home/spack/opt/spack/linux-rhel9-skylake_avx512/gcc-11.3.1/rocm-openmp-extras-5.7.1-owmri3xcla2f65bth3gfxw5g52m4j7ke/bin/flang-legacy: No such file or directory

Reply:
There should be a flang-legacy symlink in llvm-amdgpu/bin that points to the binary in rocm-openmp-extras/bin. This should only be done for releases >= 6.1 as well as the addition of flang-legacy in the components. I will have to update the patch. Thanks!